src/document.rs

```diff
@@ -123,16 +121,16 @@ pub(crate) fn extract_value_for_type(
             }
 
             Value::Object(
-                any.downcast::<PyDict>()
-                    .map(|dict| pythonize::depythonize(dict))
-                    .map_err(to_pyerr_for_type("Json", field_name, any))?
+                any.clone().downcast_into::<PyDict>()
+                    .map(|dict| pythonize::depythonize_bound(dict.clone().into_any()))
```

Comment:
clone

Reply:
There are actually two clones here. I know we can't do much about the second one, `dict.clone`, because `depythonize_bound` takes `Bound` and not a reference, but the first one is `any.clone()` and I'm not yet sure if there's a neater way to this than a clone.